### ISSUE TITLE
Multi-language support: render Japanese (CJK) text correctly and translate captions for free (e.g. English → Japanese)

### ISSUE THREAD

**Issue by irobinson010** [irobinson010/ASCII-Magic#38]
## Goal

Let text layers, captions and `ascii-magic text` work in languages beyond English. Japanese is the first target: type (or translate) text and get correct Japanese characters in ASCII art, in the terminal and the web GUI. Translation must be **free**, with no paid API as the default.

## Part 1: rendering non-Latin text (fix first; translation is useless without it)

I tested today on `main`:

| Command | Result |
|---|---|
| `ascii-magic text "こんにちは" -s block` | Each kana renders as an **empty rectangle**. The default font (DejaVu Sans Mono) has no Japanese glyphs. |
| `ascii-magic text "日本" -s figlet` | Prints **nothing at all**. A silent failure. |
| `ascii-magic text "日本語" -s box` | The border is **too short** (`┌─────┐` around `│ 日本語 │`). Width is counted with `len()`, but CJK characters take **2 terminal columns**. |

Proposed fixes:

- **Display width:** use [`wcwidth`](https://pypi.org/project/wcwidth/) (MIT, actively maintained; 0.9.1 released 2026-09-23) everywhere width is measured: box/banner, caption padding and alignment, compose layer sizes, and the colorizers' grid padding. Double-width characters need two cells in the compose grid.
- **Fonts:** fall back to a CJK font for block/small/shadow styles when the text has characters the current font lacks. Candidates are Noto Sans CJK / Noto Sans JP (both OFL-1.1). Prefer system-installed fonts and add a documented `--font` path; don't bundle multi-MB fonts in the wheel. The Noto repo warns about CFF2 variable OTFs on Windows, so use TTF there.
- **figlet:** pyfiglet 1.0.4 bundles **no kanji-capable font**. A few fonts have kana; `katakana.flf` maps romaji, not Unicode. When a figlet font can't draw the text, error out or fall back to block style instead of printing nothing. Third-party JIS FIGfonts exist (e.g. PhMajerus `phm-shinonome`) but their licenses are unconfirmed, so don't bundle them.
- **Web GUI:** the preview's HTML font stack needs a CJK fallback.

### Recommendation

**Primary: offline Argos models, run directly with CTranslate2 (optional `[translate]` extra).**

- An **en→ja** package (v1.1) and ja→en exist in the Argos package index. I verified this against `argospm-index/index.json`.
- Free, offline and private: no API key, and nothing leaves the machine.
- ⚠️ **Install weight:** the released `argostranslate` 1.11.0 wheel **hard-requires `stanza==1.10.1`, which pulls in PyTorch** (hundreds of MB, more with GPU builds). I verified this from the wheel's METADATA.
- **Proposed design:** skip the `argostranslate` package. `.argosmodel` files contain a CTranslate2 model plus SentencePiece models, so load them directly with `ctranslate2` + `sentencepiece`. Both are small, have no torch dependency, and ship wheels for Linux, macOS and Windows. Captions are short, so we don't need stanza's sentence splitting.
- **Model download:** on first use, into the user's config/cache directory. Documented and opt-in.

**Fallback: MyMemory HTTP API** (no signup; about 5k chars/day anonymous, about 50k/day with an email parameter). Useful when the offline model isn't installed. It needs network access, and the text leaves the machine, so it should be opt-in.

**Optional, bring-your-own-key:** DeepL or Google Cloud Translation, which are high quality but need signup and keys. Not a default.

### Rejected

- **Unofficial Google Translate wrappers** (`googletrans`, `deep-translator`'s Google backend): automated access is against Google's ToS, and IP bans are documented.
- **NLLB-200:** CC-BY-NC-4.0, so no commercial use.
- **`Helsinki-NLP/opus-mt-en-jap`:** trained on the Bible; unusable for captions.
- **Hosted LibreTranslate:** no free keys; self-hosting is fine but heavy, and it runs Argos anyway.
- **`staka/fugumt-en-ja`:** good quality, but CC-BY-SA and needs PyTorch. Only suitable as a user-downloaded model, never bundled.

### Proposed UX

```bash
ascii-magic text "Good morning" --translate ja -s block          # → おはようございます
ascii-magic compose --image cat.png --text "Hello" --translate ja --at top
ascii-magic translate install en ja                               # one-time model download
```

- GUI: a "Translate to…" selector on text layers and captions.
- Translation runs before rendering, and the translated text is saved in scene JSON, so re-renders are reproducible and don't need the model.

### Not yet verified (check before implementing)

- en→ja Argos model **size and quality**. There's no published EN→JA benchmark, and there's an open issue about Japanese politeness levels ([argos-translate#333](https://github.com/argosopentech/argos-translate/issues/333)). Evaluate on ~20 real captions first.
- **Argos model licenses** (per package, trained on OPUS data). Confirm redistribution terms before offering an automatic download.
- MyMemory commercial-use terms; current DeepL/Google free-tier details.
- My network proxy blocked direct access to several of these sites, so some limits came from search snippets.

## Tasks

- [ ] `wcwidth`-aware width in box/banner/caption/compose/colorize, with CJK tests
- [ ] CJK font fallback for rendered text styles, plus `--font` docs
- [ ] figlet: detect unsupported characters → clear error or fallback (no more silent empty output)
- [ ] GUI: CJK font fallback in the preview
- [ ] Evaluate the Argos en→ja model on sample captions (size, quality, license)
- [ ] `[translate]` extra: CTranslate2 + SentencePiece loader for Argos models, `ascii-magic translate install/list`
- [ ] `--translate LANG` on `text`, captions and compose text layers; GUI selector
- [ ] Optional MyMemory fallback (opt-in, network)
- [ ] Docs: supported languages, fonts, privacy notes
